src/Stratis.Bitcoin/Connection/ConnectionManager.cs

```diff
@@ -141,6 +141,22 @@ public void Initialize()
 
             this.StartNodeServer();
 
+            // if external IP address supplied this overrides all.
+            if (this.ConnectionSettings.ExternalEndpoint != null)
+            {
+                this.selfEndpointTracker.MyExternalAddress = this.ConnectionSettings.ExternalEndpoint;
+                this.selfEndpointTracker.IsMyExternalAddressFinal = true;
+            }
+            else
+            {
+                // If external IP address not supplied take first routeable bind address and set score to 10.
+                IPEndPoint nodeServerEndpoint = this.ConnectionSettings.Listen?.FirstOrDefault(x => x.Endpoint.Address.IsRoutable(false))?.Endpoint;
+                if (nodeServerEndpoint != null)
+                {
+                    this.selfEndpointTracker.UpdateAndAssignMyExternalAddress(nodeServerEndpoint, 10);
+                }
```

Comment:
We said we want else branch here which will initialise with endpoint `0.0.0.0:correctport`